docroot/sites/facilities.uiowa.edu/modules/facilities_core/src/BuildingItemProcessor.php

```diff
@@ -30,6 +31,124 @@ class BuildingItemProcessor extends EntityItemProcessorBase {
     'field_building_lactation_rooms' => 'lactationRooms',
     'field_building_latitude' => 'latitude',
     'field_building_longitude' => 'longitude',
+    'field_building_coordinators' => 'buildingCoordinators',
+    'field_building_ca_manager' => 'custodialAssistantManagerFullName',
+    'field_building_m_manager' => 'maintenanceManagerFullName',
   ];
 
+  /**
+   * Process the field_building_coordinators array.
+   */
+  public static function process($entity, $record): bool {
+    // Mapping manager fields from coordinators array.
+    if ($entity->hasField('field_building_ca_manager') && isset($record->buildingCoordinators[0]->custodialAssistantManagerFullName)) {
+      if ($entity->get('field_building_ca_manager')->value !== $record->buildingCoordinators[0]->custodialAssistantManagerFullName) {
+        $entity->set('field_building_ca_manager', $record->buildingCoordinators[0]->custodialAssistantManagerFullName);
+      }
+    }
+    if ($entity->hasField('field_building_m_manager') && isset($record->buildingCoordinators[0]->maintenanceManagerFullName)) {
+      if ($entity->get('field_building_m_manager')->value !== $record->buildingCoordinators[0]->maintenanceManagerFullName) {
+        $entity->set('field_building_m_manager', $record->buildingCoordinators[0]->maintenanceManagerFullName);
+      }
+    }
+
+    $coordinator_array = [];
+    // Check if coordinator(s) exist, if so create Paragraph(s) and save.
+    if (isset($record->buildingCoordinators[0]->mainFullName)) {
+      $main_coordinator = Paragraph::create([
+        'type' => 'uiowa_building_coordinators',
+        'field_b_coordinator_department' => $record->buildingCoordinators[0]->mainDepartment,
+        'field_b_coordinator_email' => $record->buildingCoordinators[0]->mainCampusEmail,
+        'field_b_coordinator_is_primary' => TRUE,
+        'field_b_coordinator_job_title' => $record->buildingCoordinators[0]->mainJobTitle,
+        'field_b_coordinator_name' => $record->buildingCoordinators[0]->mainFullName,
+        'field_b_coordinator_phone_number' => $record->buildingCoordinators[0]->mainCampusPhone,
+      ]);
+      $main_coordinator->save();
+      $main_array = [
+        'target_id' => $main_coordinator->id(),
+        'target_revision_id' => $main_coordinator->getRevisionId(),
+      ];
+      $coordinator_array[] = $main_array;
+    }
+
+    if (isset($record->buildingCoordinators[0]->alternateFullName1)) {
```

Comment:
This would reduce the lines of code quite a bit. Not necessarily more performant though.

```php
    $alternates = ['1', '2', '3', '4'];

    foreach ($alternates as $alt_number) {
      if (isset($record->buildingCoordinators[0]->{'alternateFullName' . $alt_number})) {
        $alternate_coordinator = Paragraph::create([
          'type' => 'uiowa_building_coordinators',
          'field_b_coordinator_department' => $record->buildingCoordinators[0]->{'alternateDepartment' . $alt_number},
          'field_b_coordinator_email' => $record->buildingCoordinators[0]->{'alternateCampusEmail' . $alt_number},
          'field_b_coordinator_is_primary' => FALSE,
          'field_b_coordinator_job_title' => $record->buildingCoordinators[0]->{'alternateJobTitle' . $alt_number},
          'field_b_coordinator_name' => $record->buildingCoordinators[0]->{'alternateFullName' . $alt_number},
          'field_b_coordinator_phone_number' => $record->buildingCoordinators[0]->{'alternateCampusPhone' . $alt_number},
        ]);
        $alternate_coordinator->save();
        $alt_array = [
          'target_id' => $alternate_coordinator->id(),
          'target_revision_id' => $alternate_coordinator->getRevisionId(),
        ];
        $coordinator_array[] = $alt_array;
      }
    }
```